### PR TITLE
User and Role bugs fixed

### PR DESCRIPTION
**Pull request by ajithandrewj**
## Description
The below issues are fixed
1 [412] Precondition Failed - user delete itself
2 User(s) need no update in Cisco Catalyst Center despite CHANGED

## Type of Change
- [ yes ] Bug fix
- [ no ] New feature
- [ no ] Breaking change
- [ no ] Documentation update

## Checklist
- [ yes ] My code follows the style guidelines of this project
- [ yes ] I have performed a self-review of my own code
- [ yes ] I have commented my code, particularly in hard-to-understand areas
- [ no ] I have made corresponding changes to the documentation
- [ yes ] My changes generate no new warnings
- [ yes ] I have added tests that prove my fix is effective or that my feature works
- [ yes ] New and existing unit tests pass locally with my changes
- [ no ] Any dependent changes have been merged and published in downstream modules
- [ yes ] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [ yes ] Tasks are idempotent (can be run multiple times without changing state)
- [ no ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [ yes ] Playbooks are modular and reusable
- [ no ] Handlers are used for actions that need to run on change

## Documentation
- [ yes ] All options and parameters are documented clearly.
- [ yes ] Examples are provided and tested.
- [ yes ] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/modules/user_role_workflow_manager.py [hunk 1]
```diff
@@ -3027,15 +3033,24 @@ def delete_user(self, user_params):
             - The function uses the "user_and_roles" family and the "delete_user_api" function from the Cisco Catalyst Center API.
         """
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:3034]: - response (dict): The API response from the "delete_user" function, 
                           or an error message if the operation fails.

In response can we update the above

File: plugins/modules/user_role_workflow_manager.py [hunk 2]
```diff
@@ -3027,15 +3033,24 @@ def delete_user(self, user_params):
...
         self.log("delete user with user_params: {0}".format(str(user_params)), "DEBUG")
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:3035]: self.log("Attempting to delete user with user_params: {0}".format(str(user_params)), "DEBUG")

File: plugins/modules/user_role_workflow_manager.py [hunk 3]
```diff
@@ -3027,15 +3033,24 @@ def delete_user(self, user_params):
...
-        return response
+        try:
+            response = self.dnac._exec(
+                family="user_and_roles",
+                function="delete_user_api",
+                op_modifies=True,
+                params=user_params,
+            )
+            self.log("Received API response from delete_user: {0}".format(str(response)), "DEBUG")
+            self.deleted_user.append(self.have.get("username"))
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:3044]: Don't we need to check the return value of the response?
```
      if response and isinstance(response, dict) and response.get("status") == "SUCCESS":
             log a message ?? 
            self.deleted_user.append(self.have.get("username"))
            return response

        error_msg = response.get("error_message", "Unknown error occurred during user deletion")
        self.log("User deletion failed. Error: {0}".format(error_msg), "ERROR")
        return {"error_message": error_msg}

   except Exception as e:
         log some message... ?? 
```

File: plugins/modules/user_role_workflow_manager.py (hunk 2 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:3035]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 3 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py [hunk 4]
```diff
@@ -3020,22 +3026,38 @@ def delete_user(self, user_params):
...
+                op_modifies=True,
+                params=user_params,
+            )
+
+            if response and isinstance(response, dict):
+                self.log("Received API response from delete_user: {0}".format(str(response)), "DEBUG")
+                self.deleted_user.append(self.have.get("username"))
+                return response
+
+            error_msg = response.get("error_message", "Unknown error occurred during user deletion")
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:3049]: Which user? How do we know?

File: plugins/modules/user_role_workflow_manager.py [hunk 5]
```diff
@@ -3020,22 +3026,38 @@ def delete_user(self, user_params):
...
+        try:
+            response = self.dnac._exec(
+                family="user_and_roles",
+                function="delete_user_api",
+                op_modifies=True,
+                params=user_params,
+            )
+
+            if response and isinstance(response, dict):
+                self.log("Received API response from delete_user: {0}".format(str(response)), "DEBUG")
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:3045]: which user?

File: plugins/modules/user_role_workflow_manager.py [hunk 6]
```diff
@@ -3020,22 +3026,38 @@ def delete_user(self, user_params):
...
+                return response
+
+            error_msg = response.get("error_message", "Unknown error occurred during user deletion")
+            self.log("User deletion failed. Error: {0}".format(error_msg), "ERROR")
+            return {"error_message": error_msg}
+
+        except Exception as e:
+            self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")
+            error_message = (
+                "Self-deletion attempt failed: User cannot delete themselves while logged into the session."
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:3056]: We can have a general message.. saying Exception occurred while deleting user {user_name}: {str(e)} ..

File: plugins/modules/user_role_workflow_manager.py (hunk 4 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 5 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py]: Updated the code accordingly.

**Pull request merged**